.circleci/config.yml

```diff
@@ -106,12 +106,38 @@ commands:
     steps:
       - install-bundle-dependencies:
           directory: << parameters.directory >>
+      - restore_cache:
+          keys:
+            - homebrew-cache-v4
+      - install-brew-dependency:
+          dependency_name: 'xcbeautify'
+      - install-brew-dependency:
+          dependency_name: 'swiftlint'
+      - save_cache:
+          key: homebrew-cache-v4
+          paths:
+            - /usr/local/Cellar/swiftlint/
+            - /usr/local/Cellar/xcbeautify/
+            - /Users/$USER/Library/Caches/Homebrew/
+  
+  install-brew-dependency:
+    parameters:
+      dependency_name:
+        type: string
+    steps:
       - run:
-          name: Install swiftlint
-          command: brew install swiftlint
-      - run:
-          name: Install xcbeautify
-          command: brew install xcbeautify
+          name: Install << parameters.dependency_name >>
+          command: |
+            # Link dependency in case it was found in the cache
+            brew link << parameters.dependency_name >> || true
+
+            if which << parameters.dependency_name >> > /dev/null 2>&1; then
+                echo "Skipping installation, already found."
+                exit 0;
+            fi
```

Comment:
What happens here if a dependency needs to be updated? It seems like if it's found it will just skip and not potentially update? Do we have to change the cache keys?

Reply:
Yeah, we'll just bump the cache key :)

Reply:
Could I suggest we try using a Homebrew Bundle? 🤷‍♂️ 

👉  https://github.com/Homebrew/homebrew-bundle

It creates a lock file that we could use the hash of the lock file as the cache key? And then we can update the lock file locally for new versions

Reply:
That doesn't allow specifying versions though?
> Homebrew does not support installing specific versions of a library, only the most recent one, so there is no good mechanism for storing installed versions in a .lock file.

Reply:
This is what _fastlane_'s lock file looks like - https://github.com/fastlane/fastlane/blob/master/fastlane/swift/formatting/Brewfile.lock.json

That quote is very confusing to me now 🙈

Reply:
Oh ok let me try. We can keep the rest of this PR the same, just use that as a hash 👍🏻

Reply:
Actually I just found [this](https://github.com/Homebrew/homebrew-bundle/issues/1057):
> Should we commit Brewfile.lock.json to git? #1057
> No, not really, unfortunately.

Reply:
Meh ignore that, I think having it will help a lot of our purpose here 👍🏻

Reply:
@joshdholtz done.